lib/ruby_lsp/global_state.rb

```diff
@@ -40,6 +40,12 @@ def initialize
       @supports_watching_files = T.let(false, T::Boolean)
       @experimental_features = T.let(false, T::Boolean)
       @type_inferrer = T.let(TypeInferrer.new(@index, @experimental_features), TypeInferrer)
+      @addon_settings = T.let({}, T::Hash[String, T.untyped])
+    end
+
+    sig { params(addon_name: String).returns(T.nilable(T::Hash[Symbol, T.untyped])) }
+    def settings_for_addon(addon_name)
+      @addon_settings[addon_name]
```

Comment:
Worth logging to `stderr` if the addon doesn't exist, to help with troubleshooting?

Reply:
By design, if the user didn't define settings for an addon this will return `nil`, so addons should always handle that scenario.

And given that addons may be different depending on which project you are working on, I feel like trying to validate the user's editor settings could be a bit tricky.